### PR TITLE
ci: add integration test with device farm

### PR DESCRIPTION
**Pull request by zhu-xiaowei**
- [x] PR title and description conform to [Pull Request](https://github.com/awslabs/clickstream-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Description of changes:*
1. add integration test with device farm

*How did you test these changes?*
Tested by device farm with Appium test script and verify logcat output after device farm test.

*Documentation update required?*
- [x] No
- [ ] Yes (Please include a PR link for the documentation update)

*General Checklist*
- [x] Added Unit Tests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/integration_test.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          ./gradlew assembleRelease
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'corretto'
+          cache: gradle
+      - name: Build sample android app
+        run: |
+          git clone -b integration_test https://github.com/aws-samples/clickstream-sdk-samples
```
Comment (zxkane) [inline .github/workflows/integration_test.yml:43]: Let's update the `build.gradle` to use local ARR for this test. Don't need another branch for the changes.

**Review by zxkane — CHANGES_REQUESTED** [review]
add license header for test files

File: .github/workflows/integration_test.yml (hunk 1 above)
Comment (zhu-xiaowei) [reply .github/workflows/integration_test.yml]: fixed.

**Comment by zhu-xiaowei** [comment]
> add license header for test files

added

**Review by zxkane — APPROVED** [review]

**Pull request merged**